src/htm/types/Sdr.cpp

```diff
@@ -308,6 +308,22 @@ namespace htm {
     }
 
 
+    void SparseDistributedRepresentation::addNoise2(const Real probability, Random& rng) {
+      NTA_ASSERT( probability >= 0.0f and probability <= 1.0f );
+      const ElemSparse numFlip = static_cast<ElemSparse>(size * probability);
+      if (numFlip == 0) return; 
```

Comment:
I'm trying to write an effective implementation, but this has problem with p << size. Should we bother with such cases? return/assert?